pkg/secrets/aws.go

```diff
@@ -0,0 +1,72 @@
+package secrets
+
+import (
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	// AWSSecretType represents the secret type for AWS credentials.
+	AWSSecretType string = "secrets.kanister.io/aws"
+
+	// AWSAccessKeyID is the key for AWS access key ID.
+	AWSAccessKeyID string = "access_key_id"
+	// AWSSecretAccessKey is the key for AWS secret access key.
+	AWSSecretAccessKey string = "secret_access_key"
+	// AWSSessionToken is the key for optional AWS session token.
+	AWSSessionToken string = "session_token"
+)
+
+// ValidateAWSCredentials validates secret has all necessary information
+// for AWS credentials. It also checks the secret doesn't have unnnecessary
+// information.
+//
+// Required fields:
+// - access_key_id
+// - secret_access_key
+//
+// Optional field:
+// - session_token
+func ValidateAWSCredentials(secret *v1.Secret) error {
+	if string(secret.Type) != AWSSecretType {
+		return errors.New("Secret is not AWS secret")
+	}
+	if _, ok := secret.Data[AWSAccessKeyID]; !ok {
+		return errors.New("awsAccessKeyID is a required field")
+	}
+	if _, ok := secret.Data[AWSSecretAccessKey]; !ok {
+		return errors.New("awsSecretAccessKey is a required field")
+	}
+	count := 2
+	if _, ok := secret.Data[AWSSessionToken]; ok {
+		count++
+	}
+	if len(secret.Data) > count {
```

Comment:
This check seems too restrictive in the sense that it should be possible to easily extend the usage of these secrets by adding other fields without having to go and explicitly modify code that uses these secrets. In particular, this is not _forward_ compatible. That is, in the future we decide to include other fields, suppose the role or some other metadata instead of the token, (a) it would require validation changes to allow the new definition and (b) older versions of the code would break when encountering the new types of secrets.